### PR TITLE
docs(oob-message): README for oob-messages

### PR DESCRIPTION
**Pull request by Christiantyemele**

### CODE REVIEW

File: oob-messages/README.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    };
+
+    let image = code.render::<Luma<u8>>().build();
+
+```
+## Dependencies
+- serde
+- url
+- base64
+- multibase
```
Comment (IngridPuppet) [inline oob-messages/README.md:85]: This is not necessary, we know where to find them.

File: oob-messages/README.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+// Generate QR code
+    let code = match QrCode::new(url.as_bytes()) {
+        Ok(qr_code) => qr_code,
+        Err(e) => return Err(format!("QR code generation error: {}", e)),
+    };
+
+    let image = code.render::<Luma<u8>>().build();
+
+```
```
Comment (IngridPuppet) [inline oob-messages/README.md:80]: You seem to be documenting an interface that is not public, meaning it is of no use for someone wanting to consume the plugin. 

From the `lib.rs` file, you can tell only two modules are public: https://github.com/adorsys/didcomm-mediator-rs/blob/85ebd002a8d5903d2f4b93b7f0f39bf11d8be37b/oob-messages/src/lib.rs#L4-L5

You should rather focus on things that matter from the perspective of someone that reuses the project than on how things are done internally unless there are side effects to declare.

File: oob-messages/README.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+3. Generating the inviation Url follows the following pattern ```https://<domain>/<path>?_oob=<encodedplaintextjwm>```.<br><domain> and <path> should be kept as short as possible, and the full URL should return human readable instructions when loaded in a browser. This is intended to aid new users. The _oob query parameter is required and is reserved to contain the DIDComm message string. Additional path elements or query parameters are allowed, and can be leveraged to provide coupons or other promise of payment for new users. e.g
+```
+   https://example.com/path?_oob=eyJ0eXBlIjoiaHR0cHM6Ly9kaWRjb21tLm9yZy9vdXQtb2YtYmFuZC8yLjAvaW52aXRhdGlvbiIsImlkIjoiNjkyMTJhM2EtZDA2OC00ZjlkLWEyZGQtNDc0MWJjYTg5YWYzIiwiZnJvbSI6ImRpZDpleGFtcGxlOmFsaWNlIiwiYm9keSI6eyJnb2FsX2NvZGUiOiIiLCJnb2FsIjoiIn0sImF0dGFjaG1lbnRzIjpbeyJpZCI6InJlcXVlc3QtMCIsIm1lZGlhX3R5cGUiOiJhcHBsaWNhdGlvbi9qc29uIiwiZGF0YSI6eyJqc29uIjoiPGpzb24gb2YgcHJvdG9jb2wgbWVzc2FnZT4ifX1dfQ
+```
```
Comment (IngridPuppet) [inline oob-messages/README.md:52]: It is good to remind what an OOB message is but it must remain brief. In fact, what one would expect from this documentation is for example the version of the protocol implemented, the endpoints available and their purpose, etc. Instead, you seem to be giving implementation guidelines (the project have likely followed them), but this is not the right place for that.

File: oob-messages/QR_Code.png
Comment (IngridPuppet) [inline oob-messages/QR_Code.png:1]: Move this image to a `/docs` subdirectory.

**Review by IngridPuppet — CHANGES_REQUESTED** [review]
You have undertaken to learn about OOB messages, and likely wanted to report your understanding here. However, this is not the right place for it.

You should change your perspective when writing a documentation like this and only focus on what someone interested in using the piece of software should know about it. This include how to get started with the project, what features are provided and in what way they are, as well as how the consumer's environment can be affected by loading the project (side effects).

Also, watch out typos.

File: oob-messages/README.md (hunk 3 above)
Comment (Christiantyemele) [reply oob-messages/README.md:52]: OK, let fix that

File: oob-messages/QR_Code.png
Comment (Christiantyemele) [reply oob-messages/QR_Code.png:1]: > Move this image to a `/docs` subdirectory.

:+1:

File: oob-messages/README.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+let oobmessagesplugin = OOBMessagesPlugin;
+
+// initialize mounting of the oob invitation and qr code in the storage directory set in **STORAGE_DIRPATH**.<br>
+// Or returns an error if the plugins are not valid ```PluginError```
+mount(oobmessagesplugin);
+
+// Navigate to following endpoints to see the oob invitation after calling ```routes``` function.
+// Then navigate to the following endpoints on your browser ```/```, ```/oob_url``` and ```/oob_qr``` .<br>
+// As defined the server configuration in the envronment variables.
+routes(oobmessagesplugin);
```
Comment (IngridPuppet) [inline oob-messages/README.md:33]: These routes are supposed to be registered by an axum server, you are not mentioning it. Also, can you document what we are expected to have on each endpoint? What are the parties bound by that OOB invitation?

File: oob-messages/README.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
+## oob-messages
+
+A Rust library for implementing out of band messages for DID-based applications.
+
+>Out of band messages (OOB) messages are the initiators of a didcomm communication where by sender provides which identifier in an unencrypted messages (QR-code or Invitation link) which the other party can scan with his/her edge device, hence no private information should be send in this way. The protocol used here is the version 2 ```https://didcomm.org/out-of-band/2.0/invitation```
```
Comment (IngridPuppet) [inline oob-messages/README.md:5]: What you quote should originate from another source verbatim.
```suggestion
>Out of band messages (OOB) messages are the initiators of a didcomm communication where by sender provides which identifier in an unencrypted messages (QR-code or Invitation link) which the other party can scan with his/her edge device, hence no private information should be send in this way. 

The protocol used here is the version 2 ```https://didcomm.org/out-of-band/2.0/invitation```
```

**Review by IngridPuppet — CHANGES_REQUESTED** [review]

File: oob-messages/README.md (hunk 5 above)
Comment (Christiantyemele) [reply oob-messages/README.md]: ok I've done the changes

**Review by Blindspot22 — COMMENTED** [review]
Good Job. But I don't thing shutting down the whole server is really necessary now!

File: oob-messages/README.md [hunk 6]
```diff
@@ -0,0 +1,43 @@
+## oob-messages
+
+A Rust library for implementing out of band messages for DID-based applications.
+
+>Out of band messages (OOB) messages are the initiators of a didcomm communication where by sender provides his identifier in an unencrypted messages (QR-code or Invitation link) which the other party can scan with his/her edge device, hence no private information should be send in this way. The protocol used here is the version 2 https://didcomm.org/out-of-band/2.0/invitation
+
```
Comment (IngridPuppet) [inline oob-messages/README.md]:
```suggestion

Out of band messages (OOB) messages are the initiators of a didcomm communication where by sender provides his identifier in an unencrypted messages (QR-code or Invitation link) which the other party can scan with his/her edge device, hence no private information should be send in this way. The protocol used here is the version 2 https://didcomm.org/out-of-band/2.0/invitation

```

**Review by IngridPuppet — APPROVED** [review]
LGTM

**Pull request merged**